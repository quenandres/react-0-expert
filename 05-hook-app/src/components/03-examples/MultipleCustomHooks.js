import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import './examples.css';

export const MultipleCustomHooks = () => {

    const { counter, increment, decrement } = useCounter(1);
    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`);
    const { quote, author } = !!data && data[0];

    return (
        <div>
            <h1>Breaking bad - Custom Hooks !!!</h1>
            <hr />
            {
            loading
            ?
            (
            <div className="alert alert-info text-center">
                Loading...
            </div>
            )
            :
            (
            <blockquote className="blockquote text-right">
                <p className="mb-0">{ quote }</p><br/>
                <footer className="blockquote-footer">{ author }</footer>
            </blockquote>
            )
            }

            {
                counter > 1 &&
                <button className="btn btn-primary" 
                onClick={decrement}    
                >
                    Anterior Frase
                </button>
            }
            {
                counter < 30 &&
                <button className="btn btn-primary" 
                    onClick={increment}
                >
                    Siguiente Frase
                </button>
            }   

        </div>
    )
}
