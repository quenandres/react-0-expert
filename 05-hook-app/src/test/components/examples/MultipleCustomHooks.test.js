import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('MultipleCustomHooks test', () => {

    

    test('Debe de mostrarse correctamente', () => {

        useCounter.mockReturnValue({
            counter: 1,
            increment: () => {},
            decrement: () => {},
            reset: () => {},
        });

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de mostrar la informacion', () => {

        useCounter.mockReturnValue({
            counter: 1,
            increment: () => {},
            decrement: () => {},
            reset: () => {},
        });
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Jhonatan',
                quote: 'Hola mundo'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        //Busquedas en el wrapper
        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('footer').text().trim() ).toBe('Jhonatan');
    });
});