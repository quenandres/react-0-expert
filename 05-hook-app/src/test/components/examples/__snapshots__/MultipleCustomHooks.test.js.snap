// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleCustomHooks test Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking bad - Custom Hooks !!!
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Frase
  </button>
</div>
`;
